ipykernel/kernelapp.py

```diff
@@ -261,6 +261,8 @@ def write_connection_file(self):
         """write connection info to JSON file"""
         cf = self.abs_connection_file
         self.log.debug("Writing connection file: %s", cf)
+        if os.path.exists(cf):
```

Comment:
Actually, sorry, can we instead add a debug log here that says that the file already exists and we're not writing it, and move the "Writing connection file" log after the return?

Reply:
Sure, putting log here is ambiguous

Reply:
Thanks!